src/controller/python/CHIPDeviceController-PyBind.cpp

```diff
@@ -0,0 +1,115 @@
+#include <controller/python/CHIPDeviceController-PyBind.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <memory>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+#include <system/SystemError.h>
+#include <system/SystemLayer.h>
+
+#include <inttypes.h>
+#include <net/if.h>
+
+#include "ChipDeviceController-StorageDelegate.h"
+
+
+#include <controller/python/Delegate.h>
+
+#include <app/CommandSender.h>
+#include <app/InteractionModelEngine.h>
+#include <controller/ExampleOperationalCredentialsIssuer.h>
+#include <inet/IPAddress.h>
+#include <mdns/Resolver.h>
+#include <support/BytesToHex.h>
+#include <support/CHIPMem.h>
+#include <support/CodeUtils.h>
+#include <support/DLLUtil.h>
+#include <support/logging/CHIPLogging.h>
+
+
+using namespace chip;
+using namespace chip::Ble;
+using namespace chip::Controller;
+
+namespace {
+chip::Controller::PythonPersistentStorageDelegate sStorageDelegate;
+chip::Controller::ExampleOperationalCredentialsIssuer sOperationalCredentialsIssuer;
+} // namespace
+
+namespace chip {
+namespace Controller {
+
+CHIPDeviceControllerPyBind::CHIPDeviceControllerPyBind()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    err = chip::Platform::MemoryInit();
+    SuccessOrExit(err);
+
+    deviceCommissioner = new chip::Controller::DeviceCommissioner();
+    VerifyOrExit(deviceCommissioner != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+    if (localDeviceId == chip::kUndefinedNodeId)
+    {
+        localDeviceId = kDefaultCtlrDeviceId;
+    }
+
+exit:
```

Comment:
Please do not use goto exit pattern, especially if there is no exit cleanup code.